src/Avalonia.Visuals/Matrix.cs

```diff
@@ -319,5 +319,76 @@ public static Matrix Parse(string s)
                 );
             }
         }
+
+        /// <summary>
+        /// Decomposes given matrix into transform operations.
+        /// </summary>
+        /// <param name="matrix">Matrix to decompose.</param>
+        /// <param name="decomposed">Decomposed matrix.</param>
+        /// <returns>The status of the operation.</returns>
+        public static bool TryDecomposeTransform(Matrix matrix, out Decomposed decomposed)
+        {
+            decomposed = default;
+
+            var determinant = matrix.GetDeterminant();
+            
+            if (MathUtilities.IsZero(determinant))
+            {
+                return false;
+            }
+
+            var m11 = matrix.M11;
+            var m21 = matrix.M21;
+            var m12 = matrix.M12;
+            var m22 = matrix.M22;
+
+            // Translation.
+            decomposed.Translate = new Vector(matrix.M31, matrix.M32);
+
+            // Scale sign.
+            var scaleX = 1d;
+            var scaleY = 1d;
+
+            if (determinant < 0)
+            {
+                if (m11 < m22)
+                {
+                    scaleX *= -1d;
+                }
+                else
+                {
+                    scaleY *= -1d;
+                }
+            }
+
+            // X Scale.
+            scaleX *= Math.Sqrt(m11 * m11 + m12 * m12);
+
+            m11 /= scaleX;
+            m12 /= scaleX;
+
+            // XY Shear.
+            double scaledShear = m11 * m21 + m12 * m22;
+
+            m21 -= m11 * scaledShear;
+            m22 -= m12 * scaledShear;
+
+            // Y Scale.
+            scaleY *= Math.Sqrt(m21 * m21 + m22 * m22);
+
+            decomposed.Scale = new Vector(scaleX, scaleY);
+            decomposed.Skew = new Vector(scaledShear / scaleY, 0d);
+            decomposed.Angle = Math.Atan2(m12, m11);
+
+            return true;
+        }
+
+        public struct Decomposed
```

Comment:
Should this be more specifically named? `DecomposedMatrix`

Reply:
Just seen its a subclass, so used like `Matrix.Domposed` so I think naming is good.

Reply:
Yeah, I've figured that from external user `Matrix.Decomposed` is good enough.